compiler/noirc_evaluator/src/acir/tests/mod.rs

```diff
@@ -926,3 +927,208 @@ fn do_not_overflow_with_constant_constrain_neq() {
+            b0(inputs: [{typ}; 2]):
+              lhs = array_get inputs, index u32 0 -> {typ}
+              rhs = array_get inputs, index u32 1 -> {typ}
+              "
+    )
+}
+
+// Create a ssa instruction corresponding to the operator, using v1 and v2 as operands.
+// Additional information can be added to the string,
+// for instance, "range_check 8" creates 'range_check v1 to 8 bits'
+fn generate_test_instruction_from_operator(operator: &str) -> (String, bool) {
+    let ops = operator.split(" ").collect::<Vec<_>>();
+    let op = ops[0];
+    let mut output = true;
+    let src = match op {
+        "constrain" => {
+            output = false;
+            format!("constrain lhs {} rhs", ops[1])
+        }
+        "not" => format!("result = {} lhs", op),
+        "truncate" => {
+            format!("result = truncate lhs to {} bits, max_bit_size: {}", ops[1], ops[2])
+        }
+        "range_check" => {
+            output = false;
+            format!("range_check lhs to {} bits", ops[1])
+        }
+        _ => format!("result = {} lhs, rhs", op),
+    };
+
+    if output {
+        (
+            format!(
+                "
+            {src}
+        return result
+        }}"
+            ),
+            true,
+        )
+    } else {
+        (
+            format!(
+                "
+            {src}
+        return
+        }}"
+            ),
+            false,
+        )
+    }
+}
+
+// Execute a simple operation for each operators
+// The operation is executed from SSA IR using ACVM after acir-gen
+// and also directly on the SSA IR via constant folding, by hardcoding the values
+// via get_constant_main_src() prefix.
+fn test_operators(
+    // The list of operators to test
+    operators: &[&str],
+    // the type of the input values
+    typ: &str,
+    // the values of the inputs
+    inputs: &[FieldElement],
+) {
+    let main = get_main_src(typ);
+    let num_type = match typ.chars().next().unwrap() {
+        'F' => NumericType::NativeField,
+        'i' => NumericType::Signed { bit_size: typ[1..].parse().unwrap() },
+        'u' => NumericType::Unsigned { bit_size: typ[1..].parse().unwrap() },
+        _ => unreachable!("invalid numeric type"),
+    };
+    let inputs_int = Value::array_from_iter(inputs.iter().cloned(), num_type).unwrap();
+    let inputs =
+        inputs.iter().enumerate().map(|(i, f)| (Witness(i as u32), *f)).collect::<BTreeMap<_, _>>();
+    let len = inputs.len() as u32;
+    let initial_witness = WitnessMap::from(inputs);
+
+    for op in operators {
+        let (src, with_output) = generate_test_instruction_from_operator(op);
+        let output = if with_output { Some(Witness(len)) } else { None };
+        let ssa = Ssa::from_str(&(main.to_owned() + &src)).unwrap();
+        // ssa execution
+        let result = ssa.interpret(vec![inputs_int.clone()]);
+        // acir execution
+        let result2 = execute_ssa(ssa, initial_witness.clone(), output.as_ref());
```

Comment:
```suggestion
        let acir_result = execute_ssa(ssa, initial_witness.clone(), output.as_ref());
```
just so it is clear one is from executing ssa and one is from executing acir. Can even rename `result` above to `ssa_result`